packages/material-ui-lab/src/MasonryItem/MasonryItem.js

```diff
@@ -0,0 +1,222 @@
+      boxSizing: 'inherit',
+      ...(styleProps.isSSR && { height: '100%' }),
+      ...(styleProps.isImage && !styleProps.isSSR && { height: styleProps.height }),
+    },
+    visibility: styleProps.height ? 'visible' : 'hidden',
+    gridColumnEnd: `span ${styleProps.columnSpan}`,
+    boxSizing: 'inherit',
+  };
+
+  const base = {};
+  Object.keys(theme.breakpoints.values).forEach((breakpoint) => {
+    if (styleProps.spacing[breakpoint] != null) {
+      base[breakpoint] = true;
+    }
+  });
+  const spacingValues = resolveBreakpointValues({ values: styleProps.spacing, base });
+  const transformer = createUnarySpacing(theme);
+  const styleFromPropValue = (propValue) => {
+    const gap = styleProps.height ? Number(getValue(transformer, propValue).replace('px', '')) : 0;
+    const rowSpan = styleProps.height ? Math.ceil(styleProps.height + gap) : 0;
+    return {
+      gridRowEnd: `span ${rowSpan}`,
+      paddingBottom: gap - 1,
+    };
+  };
+  styles = { ...styles, ...handleBreakpoints({ theme }, spacingValues, styleFromPropValue) };
+
+  return styles;
+};
+
+const MasonryItemRoot = styled('div', {
+  name: 'MuiMasonryItem',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    return [styles.root];
+  },
+})(style);
+
+const MasonryItem = React.forwardRef(function MasonryItem(inProps, ref) {
+  const props = useThemeProps({
+    props: inProps,
+    name: 'MuiMasonryItem',
+  });
+
+  const masonryItemRef = React.useRef(null);
+
+  const { spacing = 1 } = React.useContext(MasonryContext);
+  const isImage = (element) => {
+    return element.type === 'img';
+  };
+  const {
+    children,
+    className,
+    component = 'div',
+    columnSpan = 1,
+    height = isImage(children) ? 1 : undefined,
+    // images should have defined dimensions for browser to correctly lay out and paint.
+    // browser can't handle a large number of images without pre-defined dimensions
+    ...other
+  } = props;
+  const isSSR = !((isImage(children) && height === 1) || height === undefined);
+  const [styleProps, setStyleProps] = React.useState({
+    ...props,
+    isSSR,
+    isImage: isImage(children),
+    spacing,
+    columnSpan,
+    height: height < 0 ? 0 : height, // MasonryItems to which negative or zero height is passed will be hidden
+  });
+
+  const classes = useUtilityClasses(styleProps);
+  const resizeObserver = React.useRef(null);
+
+  React.useEffect(() => {
+    if (isImage(children) && styleProps.height === 1) {
+      const img = new Image();
+      img.src =
+        children.props.srcSet ||
+        children.props.src ||
+        children.props['data-src'] ||
+        children.props['data-srcSet'];
+      img.onload = () => {
+        setStyleProps({
+          ...styleProps,
+          height: img.height,
+        });
+      };
+    }
+  }, [children, styleProps]);
+
+  React.useEffect(() => {
+    // do not create a resize observer in case of SSR masonry
+    if (isSSR) {
+      return true;
```

Comment:
```suggestion
      return;
```

Effect create functions can only return a function or void